sphinx/texinputs/sphinx.sty

```diff
@@ -1279,8 +1288,35 @@
 }
 {%
   \endOriginalVerbatim
-  \par\unskip\@minipagefalse\endMakeFramed % from framed.sty snugshade
-  \ifspx@inframed\end{minipage}\fi
+  \ifspx@inframed
+   \egroup % finish \sphinxVerbatim@ContentsBox vbox
+   \ifdim\dimexpr\ht\sphinxVerbatim@ContentsBox+
+         \dp\sphinxVerbatim@ContentsBox+
+         \ht\sphinxVerbatim@TitleBox+
+         \dp\sphinxVerbatim@TitleBox>.9\textheight
```

Comment:
Rather than comparing to `0.9\textheight` the test should be with respect to remaining space on the page (`\pagegoal`, `\pagetotal`).  Indeed currently (independently of this PR) a nested code-block can easily spill over the bottom of the page and we could take the opportunity of this PR to improve this. But here the goal is only fix #6804 crash.

Reply:
Done at #6809